deepchem/models/torch_models/dtnn.py

```diff
@@ -0,0 +1,262 @@
+            - atom_number
+            - distance
+            - atom_membership
+            - distance_membership_i
+            - distance_membership_j
+
+        Returns
+        -------
+        output: torch.Tensor
+            Predictions of the Molecular Energy.
+
+        """
+        dtnn_embedding = self.dtnn_embedding(inputs[0])
+        if self.dropout > 0.0:
+            dtnn_embedding = nn.Dropout(self.dropout)(dtnn_embedding)
+        dtnn_step = self.dtnn_step(
+            [dtnn_embedding, inputs[1], inputs[3], inputs[4]])
+        if self.dropout > 0.0:
+            dtnn_step = nn.Dropout(self.dropout)(dtnn_step)
+        dtnn_step = self.dtnn_step([dtnn_step, inputs[1], inputs[3], inputs[4]])
+        if self.dropout > 0.0:
+            dtnn_step = nn.Dropout(self.dropout)(dtnn_step)
+        dtnn_gather = self.dtnn_gather([dtnn_step, inputs[2]])
+        if self.dropout > 0.0:
+            dtnn_gather = nn.Dropout(self.dropout)(dtnn_gather)
+        output = self.linear(dtnn_gather)
+        return output
+
+
+class DTNNModel(TorchModel):
+    """Implements DTNN models for regression.
+
+    This class implements the Directed Message Passing Neural Network (D-MPNN) [1]_.
+
+    Examples
+    --------
+    >>> import deepchem as dc
+    >>> import os
+    >>> model_dir = os.path.dirname(os.path.dirname(os.path.abspath(__file__)))
+    >>> input_file = os.path.join(model_dir, 'assets/example_DTNN.mat')
+    >>> X = dataset['X']
+    >>> y = dataset['T']
+    >>> w = np.ones_like(y)
+    >>> dataset = NumpyDataset(X, y, w, ids=None)
+    >>> n_tasks = y.shape[1]
+    >>> model = DTNNModel(n_tasks,
+    ...               n_embedding=20,
+    ...               n_distance=100,
+    ...               learning_rate=1.0,
+    ...               mode="regression")
+    >>> model.fit(dataset, nb_epoch=250)
+    >>> pred = model.predict(dataset)
+
+    References
+    ----------
+    .. [1] Schütt, Kristof T., et al. "Quantum-chemical insights from deep
+        tensor neural networks." Nature communications 8.1 (2017): 1-8.
+
+    """
+
+    def __init__(self,
+                 n_tasks,
+                 n_embedding=30,
+                 n_hidden=100,
+                 n_distance=100,
+                 distance_min=-1,
+                 distance_max=18,
+                 output_activation=True,
+                 mode="regression",
+                 dropout=0.0,
+                 **kwargs):
+        """
+        Parameters
+        ----------
+        n_tasks: int
+            Number of tasks
+        n_embedding: int, optional
+            Number of features per atom.
+        n_hidden: int, optional
+            Number of features for each molecule after DTNNStep
+        n_distance: int, optional
+            granularity of distance matrix
+            step size will be (distance_max-distance_min)/n_distance
+        distance_min: float, optional
+            minimum distance of atom pairs, default = -1 Angstorm
```

Comment:
Also, a quick suggestion, when specifying default values, use this format example:
    batch_no: int, optional (default 10)
      Number of batches of the electron's positions to be initialized.

Reply:
and remember to check mypy issues, there is an error in mypy tests

Reply:
> 

done